### PR TITLE
Set up CI with Azure Pipelines [skip ci]

### PR DESCRIPTION
**Pull request by MattSwartz**
This changes from using a prepackaged binary included in the project to using a nuget published package.

Please test the following:
- Does the solution build on your local machine
- Does Azure DevOps pass the build (yes it does)

### CODE REVIEW

File: WindowlessExample/packages.config [hunk 1]
```diff
@@ -1,9 +1,33 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="DotNetZip" version="1.12.0" targetFramework="net452" />
+  <package id="DotNetZip" version="1.13.0" targetFramework="net461" />
+  <package id="EngineIoClientDotNet" version="1.0.7" targetFramework="net461" />
```
[the remark this answers is not in the document]
Comment (MattSwartz) [reply WindowlessExample/packages.config]: A combination of things:
1. Upgrading from the .net framework caused some of these to get upgraded.
2. Everything else was Visual Studio switching from the supplied DLL to the nuget package. I'll do some additional testing to see if that's necessary or not.

File: AuthenticationExample/App.config [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
     <startup> 
-        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6.1" />
```
[the remark this answers is not in the document]
Comment (MattSwartz) [reply AuthenticationExample/App.config]: I'll attempt to downgrade for ease of use to match our our changes.

File: AuthenticationExample/AuthenticationExample.csproj [hunk 3]
```diff
@@ -36,27 +39,125 @@
   <PropertyGroup>
     <StartupObject>AuthenticationExample.App</StartupObject>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x86'">
```
[the remark this answers is not in the document]
Comment (MattSwartz) [reply AuthenticationExample/AuthenticationExample.csproj]: I had to explicitly define these to get the project to build in Azure DevOps

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+trigger:
+- master
+
+pool:
+  vmImage: 'VS2017-Win2016'
+
+variables:
+  solution: '**/*.sln'
+  buildPlatform: 'x86|x64'
+  buildConfiguration: 'Debug'
```
[the remark this answers is not in the document]
Comment (MattSwartz) [reply azure-pipelines.yml]: Up to you on that. If you see no benefit in ever using the Debug binaries then let's switch to Release. Not a huge deal either way as this project does not publish the binaries currently. Happy to switch it.

[the remark this answers is not in the document]
Comment (NinjaManatee) [reply azure-pipelines.yml]: I think it is better for it to default to Release, and if we need to, we can provide Debug binaries for debugging purposes.

[the remark this answers is not in the document]
Comment (MattSwartz) [reply azure-pipelines.yml]: Switched to release.

**Comment by sgd2z** [comment]
Closing this PR. resolved all conflicts with develop and made new release/3.9.0 branch with all the changes

**Pull request closed without merging**